packages/cli/src/platform/_lib/userAgent.ts

```diff
@@ -0,0 +1,14 @@
+import Debug from '@prisma/debug'
+import { isError } from '@prisma/internals'
+import * as Checkpoint from 'checkpoint-client'
+
+import { version as PRISMA_CLI_VERSION } from '../../../package.json'
+import { unknownToError } from './prelude'
+
+const debug = Debug('prisma:cli:platform:_lib:userAgent')
+
+export const getUserAgent = async () => {
+  const signature = await Checkpoint.getSignature().catch(unknownToError)
+  if (isError(signature)) debug(`await checkpoint.getSignature() failed silently with ${signature.message}`) // prettier-ignore
+  return `${isError(signature) ? 'error' : signature}.${PRISMA_CLI_VERSION}.cli.prisma`
```

Comment:
I think in case of signature error we should default the user-agent to something else than `error`?

Reply:
Well it's error for the signature, not the whole UA. Any other proposal? I think error for the signature calculation is a reasonable fallback. It could also be `unknown` but then we're just bike shedding, not really changing the overall approach. WDYT?

Reply:
Oh! I actually misread the code.

I think `unknown` makes sense, so in the case of a signature error the user-agent would be `unknown@1.2.3`

Reply:
@sampolahtinen Ok agreed regarding `unknown` instead of `error`. I don't think `<a>@<b>` captures enough information for us. I would prefer to use a domain name syntax? Any reason not to? My thinking is order of specificity: `<sig>.<ver>.cli.prisma`. Given `sig` is a different category of identifier, perhaps it makes sense to use `<ver>.cli.prisma@<sig>` where from a machine (the sig) different things could be executed from it.

Reply:
Agreed on a format in Slack. Realized in [`e97b0ab` (#23090)](https://github.com/prisma/prisma/pull/23090/commits/e97b0abf47e8b2e2c953de9d2a807a761f1a49bf).